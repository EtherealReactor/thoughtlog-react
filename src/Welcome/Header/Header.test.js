import HeaderComponent from "./Header";
// import { Provider } from 'react-redux'
import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
// import configureMockStore from 'redux-mock-store';
// import { MemoryRouter } from 'react-router-dom';

describe("DumbComponent === Header", () => {
    // let wrapper;
    // beforeEach(() => {
    //     wrapper = shallow(
    //         <MemoryRouter initialEntries={[ { pathname: '/', key: 'ram' } ]}>
    //             <Header />
    //         </MemoryRouter>
    //     );
    // });
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<HeaderComponent />)
    })

    it("must match snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("must have necessary dom", () => {
        expect(wrapper.find('div')).toHaveLength(3)
        expect(wrapper.find('header')).toHaveLength(1)
        expect(wrapper.find('nav')).toHaveLength(1)
        expect(wrapper.find('ul')).toHaveLength(1)
        expect(wrapper.find('li')).toHaveLength(2)
    })
    // 
    // it("must have necessary dom elements", () => {
    //     let dom = wrapper.find('Header').dive()
    //     expect(dom.find('header')).toHaveLength(1)
    //     expect(dom.find('nav')).toHaveLength(1)
    //     expect(dom.find('ul')).toHaveLength(1)
    //     expect(dom.find('li')).toHaveLength(2)
    //     expect(dom.find('Switch')).toHaveLength(1)
    //     expect(dom.find('Route')).toHaveLength(2)
    // });
});
// 
// 
// describe("SmartComponent === Header", () => {
//     const mockStore = configureMockStore();
//     let initialState = { user: { token: '', errors: [] } };
//     let store;
//     let wrapper;
//     beforeEach(() => {
//         store = mockStore(initialState);
//         wrapper = mount(
//             <MemoryRouter initialEntries={[ { pathname: '/', key: 'ram' } ]}>
//                 <Provider store={store}>
//                     <HeaderComponent />
//                 </Provider>
//             </MemoryRouter>
//         );
//     });
// 
//     it("must match snapshot", () => {
//         expect(toJson(wrapper)).toMatchSnapshot()
//     })
// 
//     it("must have necessary props", () => {
//         let instance = wrapper.find(Header).instance();
//         let props = instance.props
// 
//         // MapStateToProps
//         expect(props.user).toEqual(initialState.user);
//         expect(props.user.token).toEqual(initialState.user.token);
//         expect(props.user.errors).toEqual(initialState.user.errors);
// 
//         // MapDispatchToProps
//         expect(props.signup).toBeInstanceOf(Function);
//     });
// });