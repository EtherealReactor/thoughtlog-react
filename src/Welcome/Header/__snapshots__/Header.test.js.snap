// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DumbComponent === Header must match snapshot 1`] = `
<div>
  <header>
    <div>
      <div>
        <h1>
          <span>
            Thought
          </span>
          <span>
            Log
          </span>
        </h1>
      </div>
      <nav>
        <ul>
          <li>
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              exact={true}
              to="/sign_in"
            >
              Log in
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/sign_up"
            >
              Sign up
            </NavLink>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</div>
`;
